vault/vault_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package vault
+
+import (
+	"testing"
+
+	"github.com/armon/go-metrics"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestExtractor(t *testing.T) {
+	tests := []struct {
+		desc       string
+		in         []string
+		wantKey    []string
+		wantLabels []metrics.Label
+	}{
+		{
+			desc:    "vault.gcs.put",
+			in:      []string{"vault", "gcs", "put"},
+			wantKey: []string{"vault", "gcs"},
+			wantLabels: []metrics.Label{
+				{
+					Name:  "method",
+					Value: "put",
+				},
+			},
+		},
+		{
+			desc:    "vault.gcs.lock.lock",
+			in:      []string{"vault", "gcs", "lock", "lock"},
+			wantKey: []string{"vault", "gcs", "lock"},
+			wantLabels: []metrics.Label{
+				{
+					Name:  "method",
+					Value: "lock",
+				},
+			},
+		},
+		{
+			desc:    "vault.gcs.lock.unlock",
+			in:      []string{"vault", "gcs", "lock", "unlock"},
+			wantKey: []string{"vault", "gcs", "lock"},
+			wantLabels: []metrics.Label{
+				{
+					Name:  "method",
+					Value: "unlock",
+				},
+			},
+		},
+		{
+			// TODO: this should maybe be its own metric, "value" isnt a method
+			desc:    "vault.gcs.lock.value",
+			in:      []string{"vault", "gcs", "lock", "value"},
+			wantKey: []string{"vault", "gcs", "lock"},
+			wantLabels: []metrics.Label{
+				{
+					Name:  "method",
+					Value: "value",
+				},
+			},
+		},
+		{
+			desc:    "database.Initialize",
```

Comment:
It kinda bothers me that `Initialize` isn't `initialize` when everything else is lowercase. Is that Vault doing the capital I? Maybe we could downcase here?